classification/MLP_MIXER_Pytorch/main.py

```diff
@@ -0,0 +1,161 @@
+import os
+import time
+import importlib
+import json
+from collections import OrderedDict
+import logging
+import argparse
+import numpy as np
+import random
+import time
+from eval import plot_accuracy_epoch, plot_loss_epoch, make_heat_map
+from tqdm.notebook import tqdm
+import torch
+import torch.nn as nn
+import torch.optim
+import torch.utils.data
+import torch.backends.cudnn
+import torchvision.utils
+import torch.nn.functional as F
+from MLP_MIXER_Block import MixerBlock
+from MLP import MLPMixer
+from dataloader import get_loader
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--block_type", type=str, default="basic", required=True)
+    parser.add_argument("--depth", type=int, default=3, required=True)
+    parser.add_argument("--option", type=str, default="A")
+
+    # optim config
+    parser.add_argument("--epochs", type=int, default=160)
+    parser.add_argument("--batch_size", type=int, default=128)
+    parser.add_argument("--base_lr", type=float, default=0.1)
+    parser.add_argument("--weight_decay", type=float, default=1e-4)
+    parser.add_argument("--momentum", type=float, default=0.9)
+    parser.add_argument("--milestones", type=str, default="[80, 120]")
+    parser.add_argument("--lr_decay", type=float, default=0.1)
+
+    # run_config
+    parser.add_argument("--device", type=str, default="cpu")
+    parser.add_argument("--num_workers", type=int, default=2)
+
+    args = parser.parse_args()
+
+    model_config = OrderedDict(
+        [
+            ("input_size", args.input_size),
+            ("patch", args.patch),
+            ("dimension", args.dimension),
+            ("in_channels", args.in_channels),
+        ]
+    )
+
+    optim_config = OrderedDict(
+        [
+            ("epochs", args.epochs),
+            ("batch_size", args.batch_size),
+            ("base_lr", args.base_lr),
+            ("weight_decay", args.weight_decay),
+            ("momentum", args.momentum),
+            ("milestones", json.loads(args.milestones)),
+            ("lr_decay", args.lr_decay),
+        ]
+    )
+
+    data_config = OrderedDict(
+        [
+            ("dataset", "CIFAR10"),
+        ]
+    )
+    run_config = OrderedDict(
+        [
+            ("device", args.device),
+            ("num_workers", args.num_workers),
+        ]
+    )
+
+    config = OrderedDict(
+        [
+            ("model_config", model_config),
+            ("optim_config", optim_config),
+            ("data_config", data_config),
+            ("run_config", run_config),
+        ]
+    )
+
+    return config
+
+
+config = parse_args()
+
+model = REPVGG(

```

Comment:
I don't understand. Why have you used the RepVGG model for MLP mixer?

Reply:
since I edited my main file from previous repvgg model it remained unedited would correct my apologies .